### PR TITLE
[go-gen] Infer which endpoints to generate

### PR DESCRIPTION
**Pull request by c-harding**
Don’t specify which endpoints to generate, as this can fully be inferred from the templefile being passed in

### CODE REVIEW

File: src/main/scala/temple/builder/project/ProjectBuilder.scala [hunk 1]
```diff
@@ -26,29 +26,21 @@ import temple.generate.target.openapi.OpenAPIGenerator
...
-            case Endpoint.Read   => set - CRUD.Read
-            case Endpoint.Update => set - CRUD.Update
-            case Endpoint.Delete => set - CRUD.Delete
-          }
-      }
-
-    endpoints ++
-    when(service hasMetadata Metadata.ServiceEnumerable) { List } ++
-    when(service hasMetadata Metadata.ServiceAuth) { Identify }
+  def endpoints(block: AttributeBlock[_]): SortedSet[CRUD] = {
```
Comment (jaylees14) [inline src/main/scala/temple/builder/project/ProjectBuilder.scala:34]: This feels like an unnecessary rewrite, but sure.

File: src/test/scala/temple/builder/ServerBuilderTest.scala [hunk 2]
```diff
@@ -62,54 +61,13 @@ class ServerBuilderTest extends FlatSpec with Matchers {
     )
   }
 
-  it should "build a correct simple ServiceRoot with no endpoints" in {
-    val serviceRoot: ServiceRoot = BuilderTestData.simpleTemplefile.allServicesWithPorts.head match {
-      case (name, service, port) =>
-        ServerBuilder
```
Comment (jaylees14) [inline src/test/scala/temple/builder/ServerBuilderTest.scala:68]: Why has this test gone?

Comment (jaylees14) [reply src/test/scala/temple/builder/ServerBuilderTest.scala:68]: Surely we should still be testing a `Templefile` where all endpoints are omitted?

**Review by jaylees14 — APPROVED** [review]

**Review by Lewiky — APPROVED** [review]
Ok cool

**Review by Lewiky — APPROVED** [review]
Ok cool

**Pull request merged**